common/src/PCLPointCloud2.cpp

```diff
@@ -112,28 +105,27 @@ pcl::PCLPointCloud2::concatenate (pcl::PCLPointCloud2 &cloud1, const pcl::PCLPoi
         ++i;
         continue;
       }
-      if (cloud2.fields[j].name == "_")
+      else if (cloud2.fields[j].name == "_")
```

Comment:
No need of `else` here

Reply:
True.
By the way: how de we know that ``PCLPointCloud2::fields`` of cloud1 and 2 are sorted in the same order, so that it's either ``cloud1.fields[i].name == "_" ``, ``cloud2.fields[j].name == "_"`` or ``cloud1.fields[i] == cloud2.fields[j]`` ?? And if none of the three then there must be an error?

Reply:
That's the constraint that we as library implementer are putting since the cases where we need to disambiguate otherwise are too complicated.

The simple case of reordering `_ y rgb` and `rgb y _` will require comparing across all fields will make the algorithm much slower than linear time and might involve memory allocation.

Reply:
I keep proposing something here:
if I undestand correcly: ``PCLPointField`` describes data store into ``PCLPointCloud2::data``.
Why not creating a struct that contains a variant of all the possible types and the name of the field? And then ``PCLPointCloud2::data`` is a vector of this struct

Reply:
`std::variant` is a good idea. However, this discussion is about the names (the labels to the data), not the data-types inside (associated with the data). Essentially, this is a very simple map, implemented as an array.

There are standard names, `x, y, z, rgb, rgba`, etc. but the user-defined names can be anything. The order of names is also unknown since 2 point clouds can be combined to add the fields, per point.

Your proposal is attacking a different problem.